pkg/broker/provision_subscriber.go

```diff
@@ -50,6 +51,7 @@ func (p *ProvisionWorkSubscriber) Subscribe(msgBuffer <-chan WorkMsg) {
 			msg := <-msgBuffer
 			var pmsg *ProvisionMsg
 			var extCreds *apb.ExtractedCredentials
+			metrics.RemoveProvisionJob()
```

Comment:
Not entirely sure this is a valid state. We're not really removing the provision job, I guess remove is just the reference count so I guess that's ok.